charts/victoria-metrics-k8s-stack/CHANGELOG.md

```diff
@@ -1,7 +1,9 @@
 ## Next release
 
-- Updated dashboards and rules
-- updated common dependency 0.0.39 -> 0.0.42
+- update dashboards and rules
+- update common dependency 0.0.39 -> 0.0.42
+- set default vmalertmanager image to 0.28.1
+- add ability to disable vmselect or vminsert by setting replicaCount to 0. See [this issue](https://github.com/VictoriaMetrics/helm-charts/issues/2059).
```

Comment:
It's a breaking change for users who have already set `replicaCount=0` already, as the update will cause vm-operator to remove the related resources.

Reply:
added vmselect.enabled and vminsert.enabled parameters instead

Reply:
IMO, enabled flag would be an overkill (although I suggested it in the beginning).
For users who have set VMSelect replicas to 0, it would just delete the vmselect sts right ?
Also setting the replicas to 0 puts the VMCluster in failed state which isnt ideal.

Reply:
neither me nor @Haleygo were able to reproduce a cluster failure. it should be fixed in operator if there is an issue. could you please share operator logs, chart parameters to identify an error?

Reply:
Reproducing is easy.
* Deploy a VMCluster with VMSelect running 0 replicas
* Cluster should deploy fine with status either `operational` / `failed`
* Once deployed make any change to VMStorage, VMInsert configuration which will induce a rollout of both those components.

You will see the below behavior
* The change will rollout to VMStorage pods
* The change will never rollout to the VMInset pods as the operator will keep finding VMSelect pods before rolling out VMInsert
* Eventually the operation will timeout and cluster would end up in failed state
* Now if you update the VMSelect replicas to 1, the operator will find what it was looking for and eventually rollout the changes for VMInsert as well

Reply:
I wasn't able to reproduce, I've checked code, in a latest release operator skips vmselect if replicaCount is 0